File: AC/ABC-TDD/160/c.go
package main

import "fmt"

func main() {
	var k, n int
	fmt.Scan(&k, &n)
	a := make([]int, n)
	for i := 0; i < n; i++ {
		fmt.Scan(&a[i])
	}
	mid := k - a[n-1] + a[0]
	for i := 0; i < n; i++ {
		if i == 0 {
			continue
		}
		tmp := a[i] - a[i-1]
		if mid < tmp {
			mid = tmp
		}
	}
	fmt.Println(k - mid)
}

File: AC/ABC-TDD/160/c.py
k, n = map(int, input().split())
a = list(map(int, input().split()))
mid = k-a[n-1]+a[0]
for i in range(0, n):
    if i == 0:
        continue
    tmp = a[i]-a[i-1]
    if mid < tmp:
        mid = tmp
print(k-mid)
